pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -412,6 +413,10 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 			} else if infinispan.GetExposeType() == infinispanv1.ExposeTypeLoadBalancer {
 				// Waiting for LoadBalancer cloud provider to update the configured hostname inside Status field
 				if exposeAddress = kubernetes.GetExternalAddress(externalService); exposeAddress == "" {
+					if !helpers.HasLBFinalizer(externalService) {
```

Comment:
what is the relation between the LB cleanup finalizer & LB not being supported?
It seems that LB finalizer is something you add to the CR
```
 finalizers:
  - service.kubernetes.io/load-balancer-cleanup
```